### PR TITLE
sdk/log: Exporter has to be concurrent-safe

### PR DESCRIPTION
**Pull request by pellared**
Fixes https://github.com/open-telemetry/opentelemetry-go/issues/5178

SimpleProcessor does not need a mutex if a Exporter is required to be concurrent-safe.

### CODE REVIEW

File: sdk/log/simple_test.go [hunk 1]
```diff
@@ -69,30 +68,6 @@ func TestSimpleProcessorForceFlush(t *testing.T) {
 	require.True(t, e.forceFlushCalled, "exporter ForceFlush not called")
 }
 
-func TestSimpleProcessorConcurrentSafe(t *testing.T) {
```
Comment (MrAlias) [inline sdk/log/simple_test.go:72]: Why is this removed? The processor needs to ensure its methods are concurrent safe.

Comment (pellared) [reply sdk/log/simple_test.go:72]: Why if exporter is required to be concurrent safe?

Comment (pellared) [reply sdk/log/simple_test.go:72]: Should I test it with a concurrent -safe exporter?

Comment (MrAlias) [reply sdk/log/simple_test.go:72]: I'm not sure how the exporter being required to be concurrent safe removes the requirements that the simple log processor needs to also be concurrent safe. Can you explain?

Comment (pellared) [reply sdk/log/simple_test.go:72]: OK I will update to have a test which uses a concurrent-safe exporter.

Comment (MrAlias) [reply sdk/log/simple_test.go:72]: [This](https://github.com/open-telemetry/opentelemetry-go/blob/bc9c928abd1129d8a8f301b53c15472fa8624d2f/sdk/log/exporter.go#L57-L63) isn't concurrent safe?!

Comment (pellared) [reply sdk/log/simple_test.go]: 5bc5cae8 😄

**Review by MrAlias — APPROVED** [review]
